account_credit_control/models/credit_control_run.py

```diff
@@ -142,7 +158,11 @@ def generate_credit_lines(self):
         self._generate_credit_lines()
         return True
 
-    @api.multi
+    def unlink(self):
+        # Ondelete cascade don't check unlink lines restriction
```

Comment:
I don't get why is this needed. You should put `ondelete="cascade"` on child many2one, not on the one2many.

Reply:
I want to check this: https://github.com/OCA/account-financial-tools/pull/695/files#diff-de576053687d9842c8f5b25c790023fbR201